util/retry/retry.go

```diff
@@ -0,0 +1,57 @@
+package retry
+
+import (
+	"net"
+	"net/url"
+	"strings"
+
+	argoerrs "github.com/argoproj/argo/errors"
+	apierr "k8s.io/apimachinery/pkg/api/errors"
+)
+
+// IsFatalKubeAPIError returns if the error is a non-retryable kubernetes error
+func IsFatalKubeAPIError(err error) bool {
+	// get original error if it was wrapped
+	err = argoerrs.Cause(err)
+
+	if apierr.IsNotFound(err) || apierr.IsForbidden(err) {
+		return true
+	}
+	return false
+}
+
+// IsRetryableNetworkError returns whether or not the error is a retryable network error
+func IsRetryableNetworkError(err error) bool {
```

Comment:
This is currently unused, and probably needs to be tweaked -- was laying the ground work for retrying artifact load/save errors.

Reply:
nice